session/session.go

```diff
@@ -1169,10 +1169,77 @@ func (s *session) PrepareStmt(sql string) (stmtID uint32, paramCount int, fields
 	return prepareExec.ID, prepareExec.ParamCount, prepareExec.Fields, nil
 }
 
+// CachedPlanExec short path currently ONLY for cached "point select plan" execution
+func (s *session) CachedPlanExec(ctx context.Context,
+	stmtID uint32, prepared *ast.Prepared, args []types.Datum) (sqlexec.RecordSet, error) {
+	// compile ExecStmt
+	startTime := time.Now()
+	is := executor.GetInfoSchema(s)
+	execAst := &ast.ExecuteStmt{ExecID: stmtID}
+	if err := executor.ResetContextOfStmt(s, execAst); err != nil {
+		return nil, err
+	}
+	execAst.BinaryArgs = args
+	execPlan, err := plannercore.OptimizeExecStmt(ctx, s, execAst, is)
+	if err != nil {
+		return nil, err
+	}
+	stmt := &executor.ExecStmt{
+		InfoSchema:  is,
+		Plan:        execPlan,
+		StmtNode:    prepared.Stmt,
+		Ctx:         s,
+		OutputNames: execPlan.OutputNames(),
+	}
+	s.GetSessionVars().DurationCompile = time.Since(startTime)
+	stmt.Text = prepared.Stmt.Text()
+	s.GetSessionVars().StmtCtx.OriginalSQL = stmt.Text
+	logQuery(stmt.OriginText(), s.sessionVars)
+
+	// run ExecStmt
+	rs, err := stmt.GetPointRecord(ctx, is, s)
+	s.txn.changeToInvalid()
+	return rs, err
+}
+
+// IsCachedExecOk check if we can execute using plan cached in prepared structure
+// Be careful for the short path, current precondition is ths cached plan satisfying
+// IsPointGetWithPKOrUniqueKeyByAutoCommit
+func (s *session) IsCachedExecOk(ctx context.Context, prepared *ast.Prepared) (bool, error) {
+	if prepared.CachedPlan != nil {
+		plan := prepared.CachedPlan.(plannercore.Plan)
+		ok, err := plannercore.IsPointGetWithPKOrUniqueKeyByAutoCommit(s, plan)
+		if err != nil {
+			return false, err
+		}
+		if ok {
+			return true, nil
+		}
+		prepared.CachedPlan = nil
+		logutil.Logger(ctx).Error("currently only point get " +
```

Comment:
Maybe the following one is better:
```go
fmt.Sprintf(
    "only Point_Get plan can be cached in the prepare statement, invalidate the cached plan: %T",
    prepared.CachedPlan,
)
```

Reply:
auto commit mode check keep only for now